dev/tools/test/localization/gen_l10n_test.dart

```diff
@@ -0,0 +1,720 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:path/path.dart' as path;
+
+import '../../localization/gen_l10n.dart';
+import '../../localization/localizations_utils.dart';
+
+import '../common.dart';
+
+final String defaultArbPathString = path.join('lib', 'l10n');
+const String defaultTemplateArbFileName = 'app_en_US.arb';
+const String defaultOutputFileString = 'output-localization-file';
+const String defaultClassNameString = 'AppLocalizations';
+const String singleMessageArbFileString = '''{
+  "title": "Stocks",
+  "@title": {
+    "description": "Title for the Stocks application"
+  }
+}''';
+
+const String esArbFileName = 'app_es.arb';
+const String singleEsMessageArbFileString = '''{
+  "title": "Acciones"
+}''';
+
+void _standardFlutterDirectoryL10nSetup(FileSystem fs) {
+  final Directory l10nDirectory = fs.currentDirectory.childDirectory('lib').childDirectory('l10n')
+    ..createSync(recursive: true);
+  l10nDirectory.childFile(defaultTemplateArbFileName)
+    .writeAsStringSync(singleMessageArbFileString);
+  l10nDirectory.childFile(esArbFileName)
+    .writeAsStringSync(singleEsMessageArbFileString);
+}
+
+void main() {
+  MemoryFileSystem fs;
+
+  setUp(() {
+    fs = MemoryFileSystem(
+      style: Platform.isWindows ? FileSystemStyle.windows : FileSystemStyle.posix
+    );
+  });
+
+  group('LocalizationsGenerator setters:', () {
+    test('happy path', () {
+      _standardFlutterDirectoryL10nSetup(fs);
+      expect(() {
```

Comment:
You don't need the try-catch with fail here, just pass a closure to expect and the test framework will catch unexpected errors:

```
expect(() {
   final LocalizationsGenerator generator = LocalizationsGenerator(fs);
    generator.initialize(  ...  );
}, returnsNormally)
```